static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -83,32 +87,18 @@ const getTargets = (
 };
 
 const isCompatibleWithEpic = (page: Object): boolean =>
-    page.contentType === 'Article' && !page.isMinuteArticle;
+    page.contentType === 'Article' &&
+    !page.isMinuteArticle &&
+    isArticleWorthAnEpicImpression(page, defaultExclusionRules);
 
 const shouldShowReaderRevenue = (
     showToContributorsAndSupporters: boolean = false
-): boolean => {
-    const isMasterclassesPage = config
-        .get('page.keywordIds', '')
-        .includes('guardian-masterclasses/guardian-masterclasses');
-
-    return (
-        (userShouldSeeReaderRevenue() || showToContributorsAndSupporters) &&
-        !isMasterclassesPage &&
-        !config.get('page.shouldHideReaderRevenue')
-    );
-};
-
-const isEpicDisplayable = (): boolean => {
-    const page = config.get('page');
-    if (!page) {
-        return false;
-    }
-    return isCompatibleWithEpic(page) && shouldShowReaderRevenue();
-};
+): boolean =>
```

Comment:
I moved the masterclasses logic into the general `epic-exclusion-rules.js` functionality